qa/tasks/cephfs/test_admin.py

```diff
@@ -2283,7 +2283,7 @@ def test_with_health_warn_oversize_cache(self):
         errmsg = 'mds_cache_oversized'
         self.negtest_ceph_cmd(args=f'mds fail {active_mds_name}',
                               retval=1, errmsgs=errmsg)
-        self.run_ceph_cmd(f'mds fail {self.fs.name} --yes-i-really-mean-it')
+        self.run_ceph_cmd(f'mds fail {active_mds_name} --yes-i-really-mean-it')
```

Comment:
I talked about it in standup -- mds fail returns zero in such a scenario. Will add a fix for this on same PR.

Reply:
@batrick @vshankar @gregsfortytwo

I've fixed the return value issue with `mds fail` (https://tracker.ceph.com/issues/65865) and errmsg issue with `mds fail` (https://tracker.ceph.com/issues/65875) and also added a test for both.

But just now I recalled that returning zero in case of `mds fail` is done intentionally and deliberately for the sake of idempotency. So we have 2 possibilities.

1. Either return zero and modify QA code to check stderr for every command and halt the test if output on stderr wasn't expected.
2. Or forsake idempotecy and do not return zero. This will automatically halt tests.

I think the preference lies with first one because forsaking idempotency is not an option. And therefore I should delete the commits that fixes "mds fail return value" issue and instead modify run_ceph_cmd() to check stderr and halt if stderr has any extra values.

Reply:
Since then I've removed src and test code for "mds fail return value" issue and added these two commits to avoid same issues.

1. https://github.com/ceph/ceph/pull/57333/commits/f3c781f59b4c2c1750fd470277b1f9e689b0db35
2. https://github.com/ceph/ceph/pull/57333/commits/880421e12d67cfe265cad57f2061f2956dde7412

Reply:
> @batrick @vshankar @gregsfortytwo
> 
> I've fixed the return value issue with `mds fail` (https://tracker.ceph.com/issues/65865) and errmsg issue with `mds fail` (https://tracker.ceph.com/issues/65875) and also added a test for both.
> 
> But just now I recalled that returning zero in case of `mds fail` is done intentionally and deliberately for the sake of idempotency. So we have 2 possibilities.
> 
> 1. Either return zero and modify QA code to check stderr for every command and halt the test if output on stderr wasn't expected.

Some command may not return any output and follow idempotentcy rules. What would you do about those?

Reply:
I can't think of anything other than making them print some error message on standard error stream.